sycl/test-e2e/bindless_images/array/read_write_unsampled_array.cpp

```diff
@@ -0,0 +1,457 @@
+              }
+            });
+      });
+    } catch (sycl::exception e) {
+      std::cout << "\tKernel submission failed! " << e.what() << std::endl;
+      exit(-1);
+    } catch (...) {
+      std::cout << "\tKernel submission failed!" << std::endl;
+      exit(-1);
+    }
+  }
+
+  // parallel_for 2D
+  template <int NDims, typename DType, int NChannels, typename KernelName,
+            typename = std::enable_if_t<NDims == 2>>
+  static void run_ndim_test(sycl::queue q, sycl::range<2> globalSize,
+                            sycl::range<2> localSize,
+                            syclexp::unsampled_image_handle input_0,
+                            syclexp::unsampled_image_handle input_1,
+                            syclexp::unsampled_image_handle output) {
+    using VecType = sycl::vec<DType, NChannels>;
+    try {
+      q.submit([&](sycl::handler &cgh) {
+        cgh.parallel_for<KernelName>(
+            sycl::nd_range<NDims>{globalSize, localSize},
+            [=](sycl::nd_item<NDims> it) {
+              size_t dim0 = it.get_global_id(0);
+              size_t dim1 = it.get_global_id(1);
+
+              if constexpr (NChannels >= 1) {
+                VecType px1 = syclexp::read_image_array<VecType>(
+                    input_0, int(dim0), int(dim1));
+                VecType px2 = syclexp::read_image_array<VecType>(
+                    input_1, int(dim0), int(dim1));
+
+                auto sum = VecType(
+                    bindless_helpers::add_kernel<DType, NChannels>(px1, px2));
+                syclexp::write_image_array<VecType>(output, int(dim0),
+                                                    int(dim1), VecType(sum));
+              } else {
+                DType px1 = syclexp::read_image_array<DType>(input_0, int(dim0),
+                                                             int(dim1));
+                DType px2 = syclexp::read_image_array<DType>(input_1, int(dim0),
+                                                             int(dim1));
+
+                auto sum = DType(
+                    bindless_helpers::add_kernel<DType, NChannels>(px1, px2));
+                syclexp::write_image_array<DType>(output, int(dim0), int(dim1),
+                                                  DType(sum));
+              }
+            });
+      });
+    } catch (sycl::exception e) {
+      std::cout << "\tKernel submission failed! " << e.what() << std::endl;
+      exit(-1);
+    } catch (...) {
+      std::cout << "\tKernel submission failed!" << std::endl;
+      exit(-1);
+    }
+  }
+};
+
+template <int NDims, typename DType, int NChannels,
+          sycl::image_channel_type CType, sycl::image_channel_order COrder,
+          typename KernelName>
+bool run_test(sycl::range<NDims> dims, sycl::range<NDims> localSize,
+              unsigned int seed = 0) {
+  using VecType = sycl::vec<DType, NChannels>;
+
+  sycl::queue q(dev);
+  auto ctxt = q.get_context();
+
+  // skip half tests if not supported
```

Comment:
I'd add a `static_assert` against `double` that would otherwise require `aspect:fp64`.